zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/archiver/OpenSearchRepositoryIT.java

```diff
@@ -81,6 +87,81 @@ void shouldDeleteDocuments() throws IOException {
         .isEqualTo("3");
   }
 
+  @Test
+  void shouldSetIndexLifeCycle() throws IOException {
+    // given
+    final var indexName = UUID.randomUUID().toString();
+    final var repository = createRepository();
+    testClient.indices().create(r -> r.index(indexName));
+
+    final OpensearchEngineClient engineClient = new OpensearchEngineClient(testClient);
+    final OpenSearchGenericClient genericClient =
+        new OpenSearchGenericClient(testClient._transport(), testClient._transportOptions());
+
+    retention.setEnabled(true);
+    retention.setPolicyName("operate_delete_archived_indices");
+
+    // when
+    createLifeCyclePolicy(engineClient);
+    final var result = repository.setIndexLifeCycle(indexName);
+
+    // then
+    assertThat(result).succeedsWithin(Duration.ofSeconds(30));
+
+    // Takes a while for the policy to be applied
+    Awaitility.await()
+        .atMost(3, TimeUnit.SECONDS)
+        .pollInterval(1, TimeUnit.SECONDS)
+        .until(() -> indexContainsRetentionPolicy(indexName, genericClient));
+  }
+
+  @Test
+  void shouldNotSetIndexLifecycleIfRetentionIsDisabled() throws IOException {
+    // given
+    final var indexName = UUID.randomUUID().toString();
+    final var repository = createRepository();
+    testClient.indices().create(r -> r.index(indexName));
+
+    final OpensearchEngineClient engineClient = new OpensearchEngineClient(testClient);
+    final OpenSearchGenericClient genericClient =
+        new OpenSearchGenericClient(testClient._transport(), testClient._transportOptions());
+
+    retention.setEnabled(false);
+    retention.setPolicyName("operate_delete_archived_indices");
+
+    // when
+    createLifeCyclePolicy(engineClient);
+    final var result = repository.setIndexLifeCycle(indexName);
+
+    // then
+    assertThat(result).succeedsWithin(Duration.ofSeconds(30));
+
+    // we query the index lifecycle policy a few times to ensure it's not set
+    // since it takes a while for the policy to be applied.
+    final AtomicInteger attemptCounter = new AtomicInteger(0);
```

Comment:
This is indeed a much cleaner solution, thanks 🙌

Reply:
Or as we discussed, it might be even better to make a unit test out of it, no? Then asserting that `repository.setIndexLifeCycle` succeeds, because without a valid database, it would always fail.